service/frontend/service.go

```diff
@@ -308,6 +312,9 @@ func NewConfig(
 		AdminEnableListHistoryTasks: dynamicconfig.AdminEnableListHistoryTasks.Get(dc),
 
 		MaskInternalErrorDetails: dynamicconfig.FrontendMaskInternalErrorDetails.Get(dc),
+
+		FrontendHealthCheckFailureRate: dynamicconfig.FrontendHealthCheckFailureRate.Get(dc),
+		MatchingHealthCheckFailureRate: dynamicconfig.MatchingHealthCheckFailureRate.Get(dc),
```

Comment:
Not sure those are data plane config level, I was thinking server just return the data/stats and control plane will make the decision. It probably not a big deal if it's just those two configs but I guess the deep health check logic will become more and more complicate.

Reply:
We should sync offline on this. I think the cluster to tell if itself is healthy. With return stats, this api is similar to metrics and we might updating the response frequent.